lib/fluent/plugin/out_label_router.rb

```diff
@@ -26,33 +26,74 @@ class LabelRouterOutput < BareOutput
       #record_accessor_create("log")
       #record_accessor_create("$.key1.key2")
       #record_accessor_create("$['key1'][0]['key2']")
+      desc "Emit mode. If `batch`, the plugin will emit events per labels matched."
+      config_param :emit_mode, :enum, list: [:record, :batch], default: :batch
+      desc "Sticky tags will match only one record from an event stream. The same tag will be treated the same way"
+      config_param :sticky_tags, :bool, default: true
 
       config_section :route, param_name: :routes, multi: true do
-        desc "Label definition to match record. Example: app:nginx. You can specify more values as comma separated list: key1:value1,key2:value2"
-        config_param :labels, :hash, :default => {}
-        desc "Namespaces definition to filter the record. Ignored if left empty."
-        config_param :namespace, :string, :default => ""
         desc "New @LABEL if selectors matched"
         config_param :@label, :string, :default => nil
         desc "New tag if selectors matched"
         config_param :tag, :string, :default => ""
-        desc "Emit mode. If `batch`, the plugin will emit events per labels matched."
-        config_param :emit_mode, :enum, list: [:record, :batch], default: :batch
-        desc "Sticky tags will match only one record from an event stream. The same tag will be treated the same way"
-        config_param :sticky_tags, :bool, default: true
+
+        config_section :selector, param_name: :selectors, multi: true do
+          desc "Label definition to match record. Example: app:nginx. You can specify more values as comma separated list: key1:value1,key2:value2"
+          config_param :labels, :hash, :default => {}
+          desc "List of namespace definition to filter the record. Ignored if left empty."
+          config_param :namespaces, :array, :default => [], value_type: :string
+        end
+
+        config_section :exclude, param_name: :excludes, multi: true do
+          desc "Label definition to match record. Example: app:nginx. You can specify more values as comma separated list: key1:value1,key2:value2"
+          config_param :labels, :hash, :default => {}
+          desc "List of namespace definition to filter the record. Ignored if left empty."
+          config_param :namespaces, :array, :default => [], value_type: :string
+        end
+
       end
 
+
+
       class Route
-        def initialize(selector, namespace, tag, router)
+        def initialize(selectors, excludes, tag, router)
           @router = router
-          @selector = selector
-          @namespace = namespace
+          @selectors = selectors
+          @excludes = excludes
           @tag = tag
         end
 
+        # Evaluate selectors, excludes
```

Comment:
Shouldn't we perhaps try and keep the order of the select/exclude statements somehow? It could be confusing if someone writes a select first that will be filtered out with a broader exclude. 

If not than we should highlight the fact that excludes are evaluated first.

Reply:
Refactored to use <match> sections with negate parameter